[C++] Sanitize operation ids.

fvarose opened the pull request:
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

Each operation described in a swagger spec has an ID which is used by the generator directly to define the corresponding C++ function name.
This PR makes sure `operationId`s are sanitized, so that valid C++ code is generated (no spaces, dots, etc in function names).

wing328 commented:
@fvarose thanks for the PR.

cc @ravinikam @stkrwork

stkrwork commented on modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCppCodegen.java:
```diff
@@ -128,6 +128,11 @@ public String escapeReservedWord(String name) {
 
     @Override
+    public String toOperationId(String operationId) {
+        return sanitizeName(super.toOperationId(operationId));
```
What about the case if the parameter is an empty string? If its not overwritten, it checks for that.

fvarose replied:
This case is covered by the call to the default implementation (`super.toOperationId`), which throws if the string is empty.

wing328 replied:
@fvarose what about using something similar to [RubyClientCodegen.java#L619-L635](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RubyClientCodegen.java#L619-L635)?

It just a matter of time we need to handle these edge cases.

fvarose commented:
@wing328 @stkrwork 
I've added the check for reserved keywords. Unlike the Ruby client, the empty string case is performed by the default implementation, I think it's cleaner that way and avoids code duplication.
I also added the modern C++ keywords that were missing (see [reference list](http://en.cppreference.com/w/cpp/keyword)).
